src/sat/sat_clause.cpp

```diff
@@ -187,7 +187,7 @@ namespace sat {
 #if defined(Z3DEBUG)
         if (segment >= (c_max_segments-1)) {
             size_t ptr = reinterpret_cast<size_t>(cls) & ~0xFFFFFFFFull;
-            return m_ptr2cls_offset.find(reinterpret_cast<size_t>(ptr));
+            return m_ptr2cls_offset.find(reinterpret_cast<size_t>(ptr)) + segment;
```

Comment:
Sorry to jump in, but I've got some understanding of the idea of the fix attempted in #715, and I don't understand what you are trying to achieve here. `m_ptr2cls_offset.find(reinterpret_cast<size_t>(ptr))` has the same value as `segment` (`get_segment()` inserts `i` into `m_ptr2cls_offset` and then returns it). So this line is the same as `return 2 * segment`.